resources/build/zip.inc.sh

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+#
+# This script contains zip and unzip (TODO) utilities to function for all environments.
+# unzip is available in all environments
+# zip is not available by default on Windows, but can be manually installed.
+# If zip is not available, then fall back to 7z for zipping.
+# 7z requires env SEVENZ_HOME.
+#
+# TODO: refactor with /resources/build/win/zip.inc.sh
+
+# zip/7z to create an archive with the following parameters (in order)
+# [zip filename]
+# [filename of files to exclude in the zip]
+# [list of flags to pass to zip command, 'a' or leading with single-dash]
+# [list of files to include in zip]
+function zip_files() {
+
+  # Parse parameters
+
+  # $1 for zip filename
+  local ZIP_FILE="$1"
+
+  # $2 for filename containing files to exclude
+  local EXCLUDE="$2"
+  shift 2
+  if ! test -f ${EXCLUDE}; then
+    builder_die "Filename of files to exclude '${EXCLUDE}' does not exist"
+  fi
```

Comment:
I think we could do this more cleanly with a `-x filename` pattern:

```suggestion
  shift
  local EXCLUDE=
  if [[ "$1" == -x ]] then
    EXCLUDE="$2"
    shift 2
    if ! [[ -f ${EXCLUDE} ]]; then
      builder_die "Filename of files to exclude '${EXCLUDE}' does not exist"
    fi
  fi
```

I know @ermshiperete earlier suggested shift only at the end of parameter parsing, but optional parameters negates that.

But this may all be moot... why can't we pass `-x@filename` as a flag and skip the whole exclude parameter?

Reply:
I just checked:
* 7z does not need the argument at the end (in fact it should be listed before the files)
* zip used to require the argument at the end but now allows it before files list

Reply:
@mcdurdin 
k, I'm now using `-x@filename` as a flag.

-----

I couldn't find your earlier comment about the 7z parameter `a` being implicit.
On my Windows, 7-Zip version 24.09, the command looks required (so I think I'll have to insert `a`:

7-Zip 24.09 (x64) : Copyright (c) 1999-2024 Igor Pavlov : 2024-11-29

Usage: 7z <command> [<switches>...] <archive_name> [<file_names>...] [@listfile]

<Commands>

 a : Add files to archive
 b : Benchmark
 d : Delete files from archive
 e : Extract files from archive (without using directory names)
 h : Calculate hash values for files
 i : Show information about supported formats
 l : List contents of archive
 rn : Rename files in archive
 t : Test integrity of archive
 u : Update files to archive
 x : eXtract files with full paths